src/containers/PDisk/PDisk.tsx

```diff
@@ -0,0 +1,133 @@
+import {useCallback, useEffect} from 'react';
+import {StringParam, useQueryParams} from 'use-query-params';
+import {Helmet} from 'react-helmet-async';
+
+import {
+    getPDiskData,
+    getPDiskStorage,
+    setPDiskDataWasNotLoaded,
+} from '../../store/reducers/pdisk/pdisk';
+import {setHeaderBreadcrumbs} from '../../store/reducers/header/header';
+import {getNodesList, selectNodesMap} from '../../store/reducers/nodesList';
+
+import {useAutofetcher, useTypedDispatch, useTypedSelector} from '../../utils/hooks';
+import {getSeverityColor} from '../../utils/disks/helpers';
+
+import {PageMeta} from '../../components/PageMeta/PageMeta';
+import {StatusIcon} from '../../components/StatusIcon/StatusIcon';
+import {PDiskInfo} from '../../components/PDiskInfo/PDiskInfo';
+import {InfoViewerSkeleton} from '../../components/InfoViewerSkeleton/InfoViewerSkeleton';
+
+import {PDiskGroups} from './PDiskGroups';
+import {pdiskPageCn} from './shared';
+import {pDiskPageKeyset} from './i18n';
+
+import './PDisk.scss';
+
+export function PDisk() {
+    const dispatch = useTypedDispatch();
+
+    const nodesMap = useTypedSelector(selectNodesMap);
+    const {pDiskData, groupsData, pDiskLoading, pDiskWasLoaded, groupsLoading, groupsWasLoaded} =
+        useTypedSelector((state) => state.pDisk);
+    const {NodeHost, NodeId, NodeType, NodeDC, Severity} = pDiskData;
+
+    const [{nodeId, pDiskId}] = useQueryParams({
+        nodeId: StringParam,
+        pDiskId: StringParam,
+    });
+
+    useEffect(() => {
+        dispatch(setHeaderBreadcrumbs('pDisk', {nodeId, pDiskId}));
```

Comment:
No, it should be done in `useEffect`, because you can get to this page using direct link (e.g. someone send link to problematic `PDisk` in some messenger)

Reply:
Our breadcrumbs reflect logical structure, not real user path. So `Tenant` is always inside some cluster, `PDisk` is always inside some `Node`, `VDisk` is always inside some `PDisk`, etc.